servers/user-list-search/src/server/routes/itemDelete.ts

```diff
@@ -0,0 +1,91 @@
+import { deleteFromElasticSearch } from '../../elasticsearch';
+
+import { Request, Response, NextFunction, Router } from 'express';
+import { checkSchema, Schema, validationResult } from 'express-validator';
+import { nanoid } from 'nanoid';
+import * as Sentry from '@sentry/node';
+import { serverLogger } from '@pocket-tools/ts-logger';
+
+export const router = Router();
+
+const itemDeleteSchema: Schema = {
+  traceId: {
+    in: ['body'],
+    optional: true,
+    isString: true,
+    notEmpty: true,
+  },
+  userId: {
+    in: ['body'],
+    errorMessage: 'Must provide valid userId',
+    isInt: true,
+    toInt: true,
+  },
+  itemIds: {
+    in: ['body'],
+    errorMessage: 'Must provide valid itemIds',
+    isArray: true,
+    toArray: true,
+    custom: {
+      options: (value) => {
+        if (!value.every(Number.isInteger)) {
+          throw new Error('itemIds can only contain integers');
+        }
+        return true;
+      },
+    },
+  },
+};
+
+export function validate(
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): Response {
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res
+      .status(400)
+      .json({ errors: errors.array() })
+      .setHeader('Content-Type', 'application/json');
+  }
+  next();
+}
+
+router.post(
+  '/',
+  checkSchema(itemDeleteSchema),
+  validate,
+  async (req: Request, res: Response) => {
+    const requestId = req.body.traceId ?? nanoid();
+    const userId = req.body.userId;
+    const itemIds = req.body.itemIds;
+
+    serverLogger.info('starting to delete items', {
+      requestId,
+      userId,
+      itemIds,
+    });
+
+    const listItems = itemIds.map((itemId) => {
+      return { itemId, userId };
+    });
+
+    if (listItems.length > 0) {
+      try {
+        await deleteFromElasticSearch(listItems);
```

Comment:
Does this finish pretty quickly?

Reply:
so/so. I had this as async on the node background but then the tests were flakey. Which got me thinking.. it probably should not take longer then the lambda execution timeout, and we don't want lambda to just go crazy hitting this api, we should probably wait for it to finish. It also makes the lambda responsible for retries instead of it just failing.. 

i dont hold a strong opinion on this one given the current environmet.